btf/btf.go

```diff
@@ -487,6 +531,37 @@ func fixupDatasec(types []Type, sectionSizes map[string]uint32, offsets map[symb
 	return nil
 }
 
+// fixupDatasecLayout populates ds.Vars[].Offset according to var sizes and
+// alignment. Calculate and set ds.Size.
+func fixupDatasecLayout(ds *Datasec) error {
+	var off uint32
+
+	for i, vsi := range ds.Vars {
+		v := vsi.Type.(*Var)
```

Comment:
This is not always a `Var`, I think it can be `Func` as well.

Reply:
Indeed, looks like it's datasec -> vsi -> func for kfuncs. Will just `continue` here when the vsi target is not a var, since funcs don't have a size.

Reply:
I ended up returning an error here, PTAL.